app/controllers/security_events_controller.rb

```diff
@@ -0,0 +1,34 @@
+class SecurityEventsController < ApplicationController
+  before_action -> { authorize SecurityEvent }, only: %i[index all]
+
+  def index
+    @security_events = current_user.security_events.
+                       order('issued_at DESC').
+                       page(params[:page])
+
+    if @security_events.out_of_range?
```

Comment:
When I have no security events, it appears that `#out_of_range?` is `true` (somewhat surprisingly), so this triggers an infinite redirect loop.

Maybe...

```suggestion
    if !@security_events.empty? && @security_events.out_of_range?
```

Reply:
giant facepalm... but I'm worried that a page too far out ex (1000) would be both empty and out of range... will have to noodle on this, maybe just easier to just have the page be empty